src/AdamWathan/Form/Elements/Number.php

```diff
@@ -0,0 +1,29 @@
+<?php namespace AdamWathan\Form\Elements;
+
+class Number extends Input
+{
+
+    protected $attributes = array(
+        'type' => 'number',
+    );
+
+    public function placeholder($placeholder)
+    {
+        $this->setAttribute('placeholder', $placeholder);
+        return $this;
+    }
+
+    public function defaultValue($value)
+    {
+        if (! $this->hasValue()) {
+            $this->setValue($value);
+        }
+
+        return $this;
+    }
+
+    protected function hasValue()
+    {
+        return isset($this->attributes['value']);
+    }
+}
```

Comment:
If you extend the `Text` class instead of the `Input` class, all you need to set is the `type` attribute, same way the `Email` element is setup:

https://github.com/adamwathan/form/blob/master/src/AdamWathan/Form/Elements/Email.php